web/pgadmin/tools/sqleditor/static/js/components/sections/MainToolBar.jsx

```diff
@@ -288,8 +289,21 @@ export function MainToolBar({containerRef, onFilterClick, onManageMacros}) {
   };
   const warnTxnClose = ()=>{
     if(!isInTxn() || !queryToolCtx.preferences?.sqleditor.prompt_commit_transaction) {
-      eventBus.fireEvent(QUERY_TOOL_EVENTS.FORCE_CLOSE_PANEL);
-      return;
+      if(queryToolConnCtx.connectionStatus==CONNECTION_STATUS.TRANSACTION_STATUS_ACTIVE){
+        queryToolCtx.modal.showModal(gettext('Close query tool?'), (closeModal)=>(
+          <CloseQueryTool
+            closeModal={closeModal}
+            text={gettext('Their is a ongoing transaction. Do you still want to close query tool?')}
```

Comment:
Why do we need this exactly? We already have prompt commit txn for this right?

Reply:
We are basically closing the transaction and panel with this dialog. commit dialog is basically used commit or rollback of transaction.